sys/include/net/sock.h

```diff
@@ -99,13 +98,80 @@ extern "C" {
 #endif
 
 /**
- * @brief   Common flags for @ref net_conn
+ * @brief       Common flags for @ref net_conn
+ * @name        net_sock_flags
  * @{
  */
-#define SOCK_FLAGS_REUSE_EP     (0x00000001)    /**< allow to reuse end point on bind */
+#define SOCK_FLAGS_REUSE_EP     (0x0001)    /**< allow to reuse end point on bind */
 /** @} */
 
 
+/**
+ * @brief   Special netif ID for "any interface"
+ * @todo    Use an equivalent defintion from PR #5511
+ */
+#define SOCK_ADDR_ANY_NETIF (0)
+
+/**
+ * @brief   Address to bind to any IPv4 address
+ */
+#define SOCK_IPV4_EP_ANY(netif) { .family = AF_INET, \
+                                  .addr = { .ipv4 = 0U }, \
+                                  .netif = (netif) }
+
+#if defined(SOCK_HAS_IPV6) || defined(DOXYGEN)
+/**
+ * @brief   Address to bind to any IPv6 address
+ */
+#define SOCK_IPV6_EP_ANY(netif) { .family = AF_INET6, \
```

Comment:
Same here: you can only bind to an address, not an interface. In IPv6, it is necessary to specify an interface when binding to a link-local address.